### PR TITLE
Document how to specify environment in config

### LINKED ISSUE #790: Need to document better using --env in object format and not = syntax
I plucked the following out of a medium article https://blog.flennik.com/the-fine-art-of-the-webpack-2-config-dc4d19d7f172#.yk2bx4gum

screen shot 2017-02-01 at 4 24 31 pm

### PR DESCRIPTION
**Pull request by simon04**
* Using `--env`
* Using environment variables

Fixes #790.

### CODE REVIEW

File: content/configuration/environment.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
+---
+title: Environment
+contributors:
+  - simon04
+---
+
+Eventually. you will find the need to disambiguate in your `webpack.config.js` between [development](/guides/development) and [production builds](/guides/production-build). You have (at least) two options:
```
Comment (bebraw) [inline content/configuration/environment.md]: Extra . at the beginning.

**Comment by simon04** [comment]
Thanks. Fixed.

**Comment by simon04** [comment]
Relates to #401.

File: content/configuration/environment.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+## Using `--env`
+
+The webpack CLI support specifying build environment keys via `--cli` such as `--env.production` or `--env.platform=web`. To make use of those settings, change the configuration object into a function in `webpack.config.js`:
+
+```diff
+-module.exports = {
++module.exports = function(env) {
++  return {
+    plugins: [
+      new webpack.optimize.UglifyJsPlugin({
```
Comment (KSuttle-cng) [inline content/configuration/environment.md]: Sorry, why does this need to be in Uglify?

```
    new webpack.DefinePlugin({
      'process.env.NODE_ENV': JSON.stringify('production')
    })
```
This way is referenced in https://github.com/webpack/webpack.js.org/blob/master/content/guides/production.md#node-environment-variable

**Pull request merged**